### PR TITLE
Update ILM message for out of order phases error

### LINKED ISSUE #70336: Improve human-readability of ILM timing error message
In https://github.com/elastic/elasticsearch/pull/70089 we added guardrails to protect against ILM policies being created with phase timings that would result in phases running out of order.

For example, let's say the user tries to save a policy configured with a the warm phase that runs 3 days after rollover, a cold phase that runs 2 days after rollover, and a delete phase that runs 1 day after rollover. The API would reject this policy with this error:

`
phases [cold,delete] configure a [min_age] value less than the [min_age] of [3d] for the [warm] phase, configuration: {cold=2d, delete=1d}
`

The UI team hasn't budgeted time to replicate this error-checking logic in the UI, so for the time being we'll surface the API error directly to the user. Can we make it more human-readable? For example:

`
Your policy is configured to run the cold phase (min_age: 2 days) and the delete phase (min_age: 1 day) before the warm phase (min_age: 3 days). Change the phase timing so that the phases will execute in the order of hot, warm, then cold.
`

CC @dakrone @debadair @jloleysens

### PR DESCRIPTION
**Pull request by Esduard**
Closes #70336. This new message is much more readable. One small inconvenience is that the frozen and delete phases are not metioned even though they are present in the tests. I mentioned only hot, warm and cold because they are the main page names for the "Hot-warm-cold" architecture.

### CODE REVIEW

**Comment by jakelandis** [comment]
@elasticmachine ok to test

**Comment by Esduard** [comment]
I just realized I forgot to change a test in "org.elasticsearch.xpack.ilm.IndexLifecycleRestIT.test". Will correct it soon...

**Comment by Esduard** [comment]
@cjcenizal I am having a bit of trouble making the test case for the yml file `x-pack/plugin/ilm/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/ilm/10_basic.yml` line 284

To be honest i have no idea how to write yml test files.  Would you help me out by giving a few hints or links?

I am trying to write the correct catch element but i am not able to do so. Right now the statement looks like this:

`catch: /Your policy is configured to run the delete phase (min_age\:\ 5s) before the warm phase (min_age\:\ 10s). You should change the phase timing so that the phases will execute in the order of hot, warm then cold./`

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@Esduard, you were close. That line needs to be:

`catch: /Your policy is configured to run the delete phase \(min_age\:\ 5s\) before the warm phase \(min_age\:\ 10s\). You should change the phase timing so that the phases will execute in the order of hot, warm then cold./`

It's a regex expression so the parentheses also need to be escaped.

**Comment by danhermann** [comment]
@elasticmachine ok to test

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java [hunk 1]
```diff
@@ -754,8 +754,9 @@ public void testValidatingIncreasingAges() {
                 validateMonotonicallyIncreasingPhaseTimings(Arrays.asList(hotPhase, warmPhase, coldPhase, frozenPhase, deletePhase));
 
             assertThat(err,
-                containsString("phases [cold] configure a [min_age] value less than the" +
-                    " [min_age] of [1d] for the [hot] phase, configuration: {cold=12h}"));
+                containsString("Your policy is configured to run the cold phase "+
+                    "(min_age: 12h) before the hot phase (min_age: 1d). You should change "+
+                    "the phase timing so that the phases will execute in the order of hot, warm then cold."));
```
Comment (cjcenizal) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java:759]: Minor nit: Can we add another comma after warm?

> hot, warm, then cold

**Review by cjcenizal — COMMENTED** [review]
This is so awesome! Thanks for working on this @Esduard!

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java (hunk 1 above)
Comment (Esduard) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java:759]: Of course. It's a simple replace

**Comment by Esduard** [comment]
Thanks @danhermann and @cjcenizal for the feedback! I managed to update the test with the correct regex. I also put the comma at the end of the error message.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by Esduard** [comment]
@danhermann I'm puzzled by this new error. Is it happening because the test executes code from another branch?

**Comment by danhermann** [comment]
> @danhermann I'm puzzled by this new error. Is it happening because the test executes code from another branch?

Close, it's running with some nodes that are from a previous version. You'll need to temporarily disable that test by adding:

```
tasks.named("yamlRestCompatTest").configure {
  systemProperty 'tests.rest.blacklist', [
    //TODO: remove once #75099 is back ported to 7.x
    'ilm/10_basic/Test increasing phase timings validated'
  ].join(',')
}
```

to the bottom of the `x-pack/plugin/ilm/qa/rest/build.gradle` file.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Review by danhermann — COMMENTED** [review]
@Esduard, this change looks pretty good. One thing I would suggest is a test case that has 3 or more phases with bad ages since it appears that all the current test cases have only 0, 1, or 2 phases with bad ages and there's different handling for the error message in the case of 3+.

**Comment by Esduard** [comment]
@danhermann I decided to add 2 test cases: One for 3 bad phases and another for 4. That way we have test cases for every possible ammount of bad phases. Maybe Its overkill. Feel free to remove one If you like.

**Comment by danhermann** [comment]
Thanks, @Esduard! This looks good. I'll get it merged and backported to the appropriate branches.

**Pull request merged**